dartagnan/src/main/java/com/dat3m/dartagnan/solver/caat/constraints/AcyclicityConstraint.java

```diff
@@ -0,0 +1,251 @@
+package com.dat3m.dartagnan.solver.caat.constraints;
+
+
+import com.dat3m.dartagnan.solver.caat.domain.Domain;
+import com.dat3m.dartagnan.solver.caat.misc.DenseIntegerSet;
+import com.dat3m.dartagnan.solver.caat.misc.ObjectPool;
+import com.dat3m.dartagnan.solver.caat.misc.PathAlgorithm;
+import com.dat3m.dartagnan.solver.caat.predicates.CAATPredicate;
+import com.dat3m.dartagnan.solver.caat.predicates.Derivable;
+import com.dat3m.dartagnan.solver.caat.predicates.misc.MaterializedSubgraphView;
+import com.dat3m.dartagnan.solver.caat.predicates.relationGraphs.Edge;
+import com.dat3m.dartagnan.solver.caat.predicates.relationGraphs.RelationGraph;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Sets;
+
+import java.util.*;
+
+@SuppressWarnings("unchecked")
```

Comment:
My IDE generates a lot of those warnings, especially in generic code (unchecked casts are unavoidable there).
In this class, it may only be the warning generated for `onChanged` so the annotation can be moved there.
I also suppress only unchecked casts and nothing else to not accidentally cover problems.